File: programmers/etc/level2/이진_변환_반복하기.py
import re

def solution(s):
    answer = [0, 0]
    while s != "1":
        answer[0] += 1
        answer[1] += list(s).count("0")
        s = re.sub("0", "", s)
        s = bin(len(s))[2:]

    return answer


s = "110010101001"
print(solution(s)) # [3,8]

s = "01110"
print(solution(s)) # [3,3]

s = "1111111"
print(solution(s)) # [4,1]

File: Java/programmers/etc/level2/이진변환반복하기.java
package CodeTest.Java.programmers.etc.level2;

public class 이진변환반복하기 {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while (!s.equals("1")) {
            answer[0]++;
            int n = 0;
            for (char i : s.toCharArray()) {
                if (i == '0') answer[1]++;
                else n++;
            }
            s = Integer.toBinaryString(n);
        }
        
        return answer;
    }
}
